strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/nuget/TempNupkgFile.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2019 Carlspring Consulting & Development Ltd.
+ * Copyright 2014 Dmitry Sviridov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.carlspring.strongbox.nuget;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * Nuget nupkg file representation
+ *
+ * @author sviridov
+ */
+public class TempNupkgFile implements AutoCloseable
+{
+
+    private static final String ALGORITHM_NAME = "SHA-512";
```

Comment:
I think this could be dropped and replaced with `MessageDigestAlgorithms.SHA_512` wherever necessary.